quality-assurance/test-code-standards.md

```diff
@@ -1,42 +1,58 @@
 # Test Automation Standards
 
-## C# and Java
+## Test Code
 
 * Test Automation code should adhere to the same standard as production code
 * Formatting standards should reflect standard in use by development teams (e.g. Google Code Standard, Microsoft)
 * Standard coding principles apply (e.g. SOLID, DRY)
 
 ## Unit Testing
-
-* Aim for naming consistency. Common standard is to use a When-Then name, for example, WhenTwoItemsExistsThenBothItemsAreReturned
-* Follow best practice:
-    * Tests for results not functionality
-    * One assertion per test
-    * Tests should be isolated, i.e. have no dependencies on other tests nor on order of execution
-
-## API Testing
-
-* Perform full happy/unhappy path tests at this level
-* Interactions with APIs should be abstracted into a separate service/facade, not alongside the test code
+### Test a single Method in isolation
+* Test naming should follow a convention agreed by the team. An example is **Method_Action_Assertion** e.g. ValidateObject_WhenObjectPropteryIsNull_ThrowsArgurmentException
+* Tests should be isolated, i.e. have no dependencies on other tests nor on order of execution.
+* Fast. Unit tests should take very little time to run. Milliseconds not seconds!
+* Isolated. Unit tests are standalone, can be run in isolation, and have no dependencies on any outside factors such as a file system or database.
+* Repeatable. Running a unit test should be consistent with its results,it always returns the same result if you do not change anything in between runs.
+* Self-Checking. The test should be able to automatically detect if it passed or failed without any human interaction.
+* Timely. A unit test should not take a disproportionately long time to write compared to the code being tested. If you find testing the code taking a large amount of time compared to writing the code, consider a design that is more testable.
+* Unit Test coverage should aim to cover around 80% of code.
+* Aim for one assertion per test.
```

Comment:
We previously had a long discussion about this.  I think we ended up saying a test should test one outcome, but this might involve >1 assertion

Reply:
done